### PR TITLE
Internalize SqlClient Instrumentation

### PR DESCRIPTION
**Pull request by Mpdreamz**
- Internalize `Elastic.Apm.SqlClient` 

Ship this as part of the main `Elastic.Apm` binary and remove `Elastic.Apm.SqlClient` binary/nuget package. This allows us to more easily control/standardize the default diagnostics subscribers for all integrations no matter how the agent gets loaded. 

- Add `SubscribeIncludingAllDefaults()` extension method (internal)

For now only the `ElasticApmModule` uses this but the goal is to funnel all setup through this so that we have one place for all default instrumentations to be listed.

### CODE REVIEW

**Review by gregkalapos — COMMENTED** [review]
Code LGTM.

Some related things still needed to be adjusted:
- We talk about `Elastic.Apm.SqlClient` in the docs, e.g.: https://github.com/elastic/apm-agent-dotnet/blob/main/docs/packages.asciidoc#packages - that needs to be adapted.
- Remove it from the `.sh` file which pushes to NuGet https://github.com/elastic/apm-agent-dotnet/blob/main/.ci/linux/deploy.sh#L16

**Comment by Mpdreamz** [comment]
> We talk about Elastic.Apm.SqlClient in the docs, e.g.: [main/docs/packages.asciidoc#packages](https://github.com/elastic/apm-agent-dotnet/blob/main/docs/packages.asciidoc?rgh-link-date=2023-08-28T14%3A04%3A35Z#packages) - that needs to be adapted.

👍 

> Remove it from the .sh file which pushes to NuGet [main/.ci/linux/deploy.sh#L16](https://github.com/elastic/apm-agent-dotnet/blob/main/.ci/linux/deploy.sh?rgh-link-date=2023-08-28T14%3A04%3A35Z#L16)

This should already be tackled in this PR.

**Comment by gregkalapos** [comment]
> > Remove it from the .sh file which pushes to NuGet [main/.ci/linux/deploy.sh#L16](https://github.com/elastic/apm-agent-dotnet/blob/main/.ci/linux/deploy.sh?rgh-link-date=2023-08-28T14%3A04%3A35Z#L16)
> 
> This should already be tackled in this PR.

Literally the first file under "Files changed" 😄 Totally overlooked, sorry.

**Comment by Mpdreamz** [comment]
One thing I just considered that if someone updates and has a hard reference on the `Elastic.Apm.SqlClient` package and manually adds it to subscribe. 

Need to ensure this does not cause double subscriptions

**Comment by Mpdreamz** [comment]
@gregkalapos added some protection from diagnostic subscribers from removed nuget packages accidentally subscribing in addition to the moved builtin versions

https://github.com/elastic/apm-agent-dotnet/pull/2165/files#diff-194e6f95448df8b0723031b47ec12eed4a9fffa38a6fb739b9e8cbf2b34ff375R62-R74

File: src/Elastic.Apm/ApmAgentExtensions.cs [hunk 1]
```diff
@@ -59,12 +59,20 @@ internal static IDisposable SubscribeIncludingAllDefaults(this IApmAgent agent,
 				new SqlClientDiagnosticSubscriber(),
 				new HttpDiagnosticsSubscriber()
 			};
+			// on the off chance that someone manually references old nuget packages and injects them manually
```
Comment (gregkalapos) [inline src/Elastic.Apm/ApmAgentExtensions.cs:62]: 👍

**Review by gregkalapos — APPROVED** [review]

**Comment by holterbades** [comment]
Hi, 
i just stumbled over this PR when I was searching for a solution regarding the `SqlClientDiagnosticSubscriber`. 
I recently upgraded the `Elastic.Apm` package to the current version `1.27.3` and did not imediately notice that the `Elastic.Apm.SqlClient` package version was behind on `1.23.0`. When I ran the application, the following exception occoured:

> System.MethodAccessException: "Attempt by method 'Elastic.Apm.SqlClient.SqlClientDiagnosticSubscriber.Subscribe(Elastic.Apm.IApmAgent)' to access method 'Elastic.Apm.CompositeDisposable..ctor()' failed."

Since in the documentation was still the mention of this `Elastic.Apm.SqlClient` package and no obvious note that it is no longer compatible or should not be referenced (like in the [packages.asciidoc#sqlclient](https://github.com/elastic/apm-agent-dotnet/blob/main/docs/packages.asciidoc#sqlclient)), I was unnecessarily searching for a solution.

Maybe the documentation could be more obvious and the nuget package should be [marked as deprecated](https://learn.microsoft.com/en-us/nuget/nuget-org/deprecate-packages) to make it more clear that it should not be referenced anymore.

Thank you for considering this.

**Pull request merged**